bigquery/google/cloud/bigquery/job.py

```diff
@@ -92,6 +93,14 @@ def _error_result_to_exception(error_result):
     )
 
 
+def _contains_order_by(query):
+    """Do we need to preserve the order of the query results?"""
+    if query is None:
+        return False
+
+    return re.search(r"ORDER\s+BY", query, re.IGNORECASE) is not None
```

Comment:
Please compile the pattern as a module-scope global, and use "raw" strings e.g.:

```python
_CONTAINS_ORDER_BY = re.compile(r"ORDER\s+BY", re.IGNORECASE)

...
def _contains_order_by(query):
    """Do we need to preserve the order of the query results?"""
    return query and _CONTAINS_ORDER_BY.search(query)
```